kube/src/api/subresource.rs

```diff
@@ -221,6 +222,84 @@ where
     }
 }
 
+// ----------------------------------------------------------------------------
+// Eviction subresource
+// ----------------------------------------------------------------------------
+
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
+/// Eviction body
+#[derive(Default, Clone, Serialize)]
+pub struct Eviction {
+    /// How the eviction should occur
+    pub delete_options: Option<DeleteParams>,
+    /// Metadata describing the pod being evicted
+    /// we somehow need this despite providing the name of the pod..
+    pub metadata: ObjectMeta,
```

Comment:
That feels like a pretty big footgun.. Maybe `Api::evict` could take the `name` from the `Eviction` (or the reverse)?

Reply:
yeah, we could just pave over that interface, and take a (DeleteParams, PostParams, name) and only have the Eviction struct be internal. I don't see what good the metadata would do if it's tied to a named pod anyway.

Reply:
Made an EvictParams object containing the two params objects, and hid Eviction in the body of `Resource::evict`